presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -5301,6 +5301,76 @@ public void testUnsupportedCsvTable()
     }
 
+    @Test
+    public void testCreateMaterializedView()
+    {
+        computeActual("CREATE TABLE test_customer_base WITH (partitioned_by = ARRAY['nationkey']) AS SELECT custkey, name, address, nationkey FROM customer LIMIT 10");
+        computeActual("CREATE TABLE test_orders_base WITH (partitioned_by = ARRAY['orderstatus']) AS SELECT orderkey, custkey, totalprice, orderstatus FROM orders LIMIT 10");
+
+        // Test successful create
+        assertUpdate("CREATE MATERIALIZED VIEW test_customer_mv WITH (partitioned_by = ARRAY['nationkey']) AS SELECT name, nationkey FROM test_customer_base");
+        assertTrue(getQueryRunner().tableExists(getSession(), "test_customer_mv"));
+        assertTableColumnNames("test_customer_mv", "name", "nationkey");
+
+        // Test if exists
+        assertQueryFails(
+                "CREATE MATERIALIZED VIEW test_customer_mv AS SELECT name FROM test_customer_base",
+                format(".* Materialized view '%s.%s.test_customer_mv' already exists", getSession().getCatalog().get(), getSession().getSchema().get()));
+        assertQuerySucceeds("CREATE MATERIALIZED VIEW IF NOT EXISTS test_customer_mv AS SELECT name FROM test_customer_base");
+
+        // Test partition mapping
+        assertQueryFails(
+                "CREATE MATERIALIZED VIEW test_customer_mv_no_partition AS SELECT name FROM test_customer_base",
+                ".*Unpartitioned materialized view is not supported.");
+        assertQueryFails(
+                "CREATE MATERIALIZED VIEW test_customer_mv_no_direct_partition_mapping WITH (partitioned_by = ARRAY['nationkey']) AS SELECT name, CAST(nationkey AS BIGINT) AS nationkey FROM test_customer_base",
+                format(".*Materialized view %s.test_customer_mv_no_direct_partition_mapping must have at least one partition directly defined by a base table partition.", getSession().getSchema().get()));
+
+        // Test nested
+        assertQueryFails(
+                "CREATE MATERIALIZED VIEW test_customer_nested_mv WITH (partitioned_by = ARRAY['nationkey']) AS SELECT name, nationkey FROM test_customer_mv",
+                format(".*CreateMaterializedView on a materialized view %s.%s.test_customer_mv is not supported.", getSession().getCatalog().get(), getSession().getSchema().get()));
+
+        // Test query shape
+        assertUpdate("CREATE MATERIALIZED VIEW test_customer_agg_mv WITH (partitioned_by = ARRAY['nationkey']) AS SELECT COUNT(DISTINCT name) AS num, nationkey FROM (SELECT name, nationkey FROM test_customer_base GROUP BY 1, 2) a GROUP BY nationkey");
+        assertUpdate("CREATE MATERIALIZED VIEW test_region_nation_join_mv WITH (partitioned_by = ARRAY['nationkey']) AS SELECT orders.orderstatus AS orderstatus, customer.nationkey FROM test_customer_base customer JOIN test_orders_base orders ON orders.custkey = customer.custkey");
```

Comment:
Is this correct? What would be the column mapping looks like? 
test_region_nation_join_mv.nationkey = test_customer_base.nationkey
but test_orders_base is with orderstatus as the partition key right?

Reply:
Right, revised.

Reply:
I'm curious why this will pass assert? It should fail right. It doesn't look like a legit use case.

Reply:
Suppose for the use case below, the exchange_rate is a small dimension table partitioned by 'country'. If we include country as the mv partition key, a reload of a country will trigger the refresh of impacted revenue_aggregation_mv partitions. If we don't include country as a mv partition key, a reload of country won't affect the past revenue_aggregation_mv data. Need to have this flexibility to handle different customer needs.

CREATE MATERIALIZED VIEW IF NOT EXISTS revenue_aggregation_mv
WITH (
    partitioned_by = ARRAY['ds']
)
AS SELECT
    SUM(l.legal_budget * r.to_usd) AS total_legal_budget_usd,
    country,
    ds
FROM
    revenue_aggregation_imp l
JOIN
    exchange_rate r
ON l.country = r.country
GROUP BY
    country,
    ds